services/app/src/actions/Card.tsx

```diff
@@ -40,6 +41,12 @@ export const toCard = remoteify(function toCard(a: ToCardArgs, dispatch: Redux.D
     keylist.push(a.keySuffix);
   }
 
+  // Log the end of the quest if we reach it.
+  if (a.name === 'QUEST_END') {
+    console.log('Dispatching logQuestPlay');
+    dispatch(logQuestPlay({phase: 'end'}));
```

Comment:
Out of curiosity, why did this get moved from the Quest End card to here? And, if so, would it make sense to also move the log start of play dispatch to here as well, to keep all of our card / navigation-based analytics in one spot?

Reply:
It was moved to here because logQuestPlay is a thunk action, requiring a call to `dispatch`... having it be part of the component constructor is really weird, since that should just be a view and not dispatch actions directly.

But it's a bit sloppy to have it here in `toCard` as well... I moved this to a separate `endQuest` action which seems a bit nicer.